src/bucket.ts

```diff
@@ -1233,8 +1242,18 @@ class Bucket extends ServiceObject {
       return apiFormattedRule;
     });
 
+    this.disableAutoRetryConditionallyIdempotent_(
```

Comment:
I'm not any kind of node expert but it looks like you are disabling the retry during the setMetadata operation and then reenabling it afterwards. If so, does that pose any kind of concurrency/race condition issues with the client? Also, will user configuration choices override this logic?

Reply:
There shouldn't be an issue with concurrency but I will make sure there isn't the possibility of a race condition. This is overriding the users configuration temporarily so actually the opposite of your second question.

Reply:
Gotcha. Does that mean that users can't configure the library to retry in spite of the operation being non-idempotent?

Reply:
The user can still configure the library to retry. However, for these conditionally idempotent operations if the correct precondition is not set, the library will override the user setting and turn off retries which is what the [spec](https://cloud.google.com/storage/docs/retry-strategy) notes.